src/audio/module_adapter/module/modules.c

```diff
@@ -63,46 +62,15 @@ static int modules_new(struct processing_module *mod, const void *buildinfo,
 	/* Connect loadable module interfaces with module adapter entity. */
 	/* Check if native Zephyr lib is loaded */
 
-	const struct sof_module_api_build_info *mod_buildinfo;
-
-	if (buildinfo) {
-		mod_buildinfo = buildinfo;
-	} else {
-		const struct sof_man_module *const module_entry =
-			lib_manager_get_module_manifest(module_id);
-
-		if (!module_entry) {
-			comp_err(dev, "modules_new(): Failed to load manifest");
-			return -ENODATA;
-		}
-
-		mod_buildinfo =
-			(struct sof_module_api_build_info *)
-			(module_entry->segment[SOF_MAN_SEGMENT_TEXT].v_base_addr);
-	}
```

Comment:
this breaks llext

Reply:
@softwarecki if you **really** think that you'd fixed the breakage, I'd need to test it, but from what I see I still don't see where the `buildinfo` pointer is **obtained** in the llext case. Currently it's obtained at https://github.com/thesofproject/sof/blob/6f33aee2709ac619c4913a231f40e5b27d4390a5/src/library_manager/llext_manager.c#L220 and then it's propagated all the way up. Your patches seem to break that propagation. Am I missing anything?